pkg/controllers/disruption/singlenodeconsolidation.go

```diff
@@ -83,7 +83,7 @@ func (s *SingleNodeConsolidation) ComputeCommand(ctx context.Context, disruption
 			return Command{}, scheduling.Results{}, fmt.Errorf("validating consolidation, %w", err)
 		}
 		if !isValid {
-			logging.FromContext(ctx).Debugf("abandoning single-node consolidation attempt due to pod churn, command is no longer valid, %s", cmd)
+			controllerruntime.LoggerFrom(ctx).V(3).Info("abandoning single-node consolidation attempt due to pod churn, command is no longer valid, %s", cmd)
```

Comment:
This is a good starting place I would say. We can develop more granularity and a better system for verbosity later, right now the important thing is to have logs available that do not use the Knative logging package

Reply:
Some guidelines should be decided around verbosity ie something in a loop should be V5 etc. But I don't think that work needs to be solved in this pr

Reply:
Yes, I agree. Perhaps we can have a follow up pr for that.

Reply:
The recommendation [directly from `go-logr`](https://zerokspot.com/weblog/2021/10/30/go-logr/) is that `V(0)` basically means our current "Info" level (e.g. I want to see this message all of the time, but it's not an error). `V(1)` basically means our current "Debug" level. Everything lower than that basically becomes trace logging. I'm not convinced that we want to expand into trace logging yet since I think that that can start to get highly unwieldy the further down that we go. 

We already have questions today about whether a given log is an Info() log or a Debug() log and even those questions are hard to answer. 

<img width="837" alt="Screenshot 2024-02-29 at 8 15 26 PM" src="https://github.com/kubernetes-sigs/karpenter/assets/26334334/7dda3cc8-d3d2-4549-913a-02cccb819411">

Reply:
We should do a quick sanity check to see what is done in upstream and some other project too. I'm less familiar with what the standard is out there in the community for these verbosity levels, but I'd imagine that starting from the upstream documentation based on the docs shipped from the package writers themselves is a good place to go from

Reply:
Thanks @jonathan-innis . I have updated my patch now accordingly (haven't pushed yet) but as you suggested I will try to find how other folks around k8s using verbosity and if there is some guidelines or best practices set. I will come back to this very soon.